src/engine/SpatialJoinAlgorithms.h

```diff
@@ -163,11 +174,25 @@ class SpatialJoinAlgorithms {
   // Only for the poles, the conversion will be way to large (for the longitude
   // difference). Note, that this function is expensive and should only be
   // called when needed
-  double computeDist(const AnyGeometry& geometry1,
-                     const AnyGeometry& geometry2) const;
+  double computeDist(const std::optional<size_t> geometryIndex1,
+                     const std::optional<size_t> geometryIndex2) const;
+
+  // this helper function takes an idtable, a row and a column. It then tries
+  // to parse a geometry or a geoPoint of that cell in the idtable. If it
+  // succeeds, it returns an rtree entry of that geometry/geopoint
+  std::optional<rtreeEntry> getRtreeEntry(const IdTable* idTable,
+                                          const size_t row,
+                                          const ColumnIndex col);
```

Comment:
I the `rtreeEntry` class (which should be called `RtreeEntry` , the geometry index is an `optional<size_t>` I think that is conceptually wrong, as the index always has to exist, so please make it a plain `size_t` then you get rid of a lot of the `.value()` calls.